<a href="https://colab.research.google.com/github/ikwak2/StatBook/blob/main/ch11/11_5_R_prac.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 11.5 R 연습

## <예 11-1>에 대한 카이제곱 적합도 검정 과정
### 자료의 입력

In [1]:
p<-c(713,215,22.,52)
x<-c(690,232,234,44)
p
x

[1] 713 215  22  52

[1] 690 232 234  44

여기서 p는 기대도수, x는 관측도수를 나타낸다.

### 카이제곱 검정

In [2]:
example.11.1<-chisq.test(x,p=p,rescale.p=T)
example.11.1


	Chi-squared test for given probabilities

data:  x
X-squared = 1675.9, df = 3, p-value < 2.2e-16


R에서 기본으로 제공하는 명령어 chisq.test를 이용하여 회귀분석을 실시한다. 명령문에서 괄호 안의 첫 번째는 관측도수, 두 번째는 기대도수이다. 세 번째 옵션 rescale.p=T는 두 번째 입력변수 p를 관측도수로 사용할 것인지의 여부이다. rescale=F를 입력할 경우 p에는 관측도수가 아닌 비율이 들어가야 한다. (이 예제에서는 c(9/16,3/16,3/16,1/16)) 

### 결과 해석

검정하고자 하는 가설은

$H_0$ : 실험결과는 멘델의 법칙을 잘 따른다.  
$H_1$ : 실험결과는 멘델의 법칙을 잘 따르지 않는다.

이다. 출력된 결과에서 P값이 2.2e-16이므로 5% 유의수준 하에서 귀무가설을 기각할 수 없다. 즉, 5% 유의수준 하에서 실험결과는 멘델의 법칙을 잘 따른다고 할 수 있다.

## <예 11-2>에 대한 카이제곱 독립성 검정 과정
### 자료의 입력

In [3]:
y<-matrix(c(23,10,17,296,182,472),nrow=2,byrow=T)
rownames(y)<-c("감염되었음","감염되지 않았음")
colnames(y)<-c("미접종","1회","2회")
y

미접종 1회 2회
감염되었음       23     10  17
감염되지 않았음 296    182 472


matrix 명령어를 이용하여 행렬을 생성한다. 여기서 rownames, colnames는 각각 행이름, 열이름을 지정하는 명령어이다. 굳이 입력하지 않아도 검정에는 지장이 없다.

### 카이제곱 검정

In [4]:
example.11.2<-chisq.test(y)
example.11.2


	Pearson's Chi-squared test

data:  y
X-squared = 5.6872, df = 2, p-value = 0.05822


R에서 기본으로 제공하는 명령어 chisq.test를 이용하여 회귀분석을 실시한다. 적합도 검정과 달리 행렬 자료를 입력할 경우 행변수, 열변수에 대해 카이제곱 검정을 실시한다. 

### 결과 해석
검정하고자 하는 가설은

$H_0$ : 예방접종의 실시 여부와 간염의 감염 여부는 서로 독립이다.  
$H_1$ : 예방접종의 실시 여부와 간염의 감염 여부는 서로 독립이 아니다.

이다. 출력된 결과에서 P값이 0.05822이므로 5% 유의수준 하에서 귀무가설을 기각할 수 없다. 즉, 5% 유의수준 하에서 예방접종의 실시 여부와 간염의 감염 여부는 서로 독립이라고 할 수 있다. (즉, 서로 관련이 있다고 할 수 없다.)

In [5]:
example.11.2$expected

미접종 1회   2회   
감염되었음       15.95   9.6  24.45
감염되지 않았음 303.05 182.4 464.55

각 셀의 기대도수도 확인할 수 있다.

